### DISCUSSION TITLE
consult-buffer "End of file during parsing" error

### DISCUSSION THREAD

**Discussion by JosukeBukin** [minad/consult#1127] [Q&A] [answered]
Error occures on `emacs -Q` with minimal configuration provided in the README.
```
(package-initialize)
(require 'consult)
(require 'vertico)
(vertico-mode)
(setq completion-styles '(substring basic))
```
Error seems occures on `consult--multi-enabled-sources` call, at least by trying to use debug-on-entry

I'm using `GNU Emacs 29.4 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.43, cairo version 1.18.2)`

**Comment by minad** [comment]
I cannot reproduce this. Maybe there is an installation problem?

**Reply by JosukeBukin** [reply]
Maybe, because on my other host (both are using Arch) consult-buffer is working properly

**Reply by JosukeBukin** [reply]
But both are official Emacs packages

**Reply by JosukeBukin** [reply]
I got end of file during parsing error when calling this function in consult-buffer
`consult--multi-enabled-sources((consult--source-hidden-buffer consult--source-modified-buffer consult--source-buffer consult--source-recent-file consult--source-file-register consult--source-bookmark consult--source-project-buffer-hidden consult--source-project-recent-file-hidden))`

How i can check what each argument contains using debug or similar tools?

**Reply by JosukeBukin** [reply]
I think the problem is with bookmarks function, because calling `bookmark-bmenu-list` return end of file during parsing

**Reply by JosukeBukin** [reply]
Bookmarks file `.emacs.d/bookmarks` exists

**Reply by JosukeBukin** [reply, marked as answer]
Deleting `.emacs.d/bookmarks` helps to solve issue. This problem probably causes because my bookmarks file was empty and load-bookmark function returns end of file during parsing
